// @angular/material/table/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/material/table';
import * as i2 from '@angular/common';
import * as i3 from '@angular/cdk/bidi';
import * as i4 from '@angular/platform-browser';
import * as i5 from '@angular/cdk/table';
import * as i6 from '@angular/material/core';
export const MatTableModuleNgFactory:i0.NgModuleFactory<i1.MatTableModule> = i0.ɵcmf(i1.MatTableModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(6144,i3.DIR_DOCUMENT,
          (null as any),[i4.DOCUMENT]),i0.ɵmpd(4608,i3.Directionality,i3.Directionality,
          [[2,i3.DIR_DOCUMENT]]),i0.ɵmpd(512,i2.CommonModule,i2.CommonModule,([] as any[])),
          i0.ɵmpd(512,i5.CdkTableModule,i5.CdkTableModule,([] as any[])),i0.ɵmpd(512,
              i6.CompatibilityModule,i6.CompatibilityModule,([] as any[])),i0.ɵmpd(512,
              i3.BidiModule,i3.BidiModule,([] as any[])),i0.ɵmpd(256,i6.MATERIAL_SANITY_CHECKS,
              true,([] as any[])),i0.ɵmpd(512,i6.MatCommonModule,i6.MatCommonModule,
              [[2,i6.MATERIAL_SANITY_CHECKS]]),i0.ɵmpd(512,i1.MatTableModule,i1.MatTableModule,
              ([] as any[]))]);
    });
const styles_MatTable:any[] = ['.mat-table{display:block}.mat-header-row,.mat-row{display:flex;border-bottom-width:1px;border-bottom-style:solid;align-items:center;min-height:48px;padding:0 24px}.mat-cell,.mat-header-cell{flex:1}'];
export const RenderType_MatTable:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MatTable,
    data:{}});
export function View_MatTable_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(402653184,1,{_rowPlaceholder:0}),i0.ɵqud(402653184,2,{_headerRowPlaceholder:0}),
      (_l()(),i0.ɵeld(2,16777216,(null as any),(null as any),1,(null as any),(null as any),
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(3,16384,[[2,4]],0,i5.HeaderRowPlaceholder,[i0.ViewContainerRef],(null as any),
          (null as any)),(_l()(),i0.ɵeld(4,16777216,(null as any),(null as any),1,
          (null as any),(null as any),(null as any),(null as any),(null as any),(null as any),
          (null as any))),i0.ɵdid(5,16384,[[1,4]],0,i5.RowPlaceholder,[i0.ViewContainerRef],
          (null as any),(null as any))],(null as any),(null as any));
}
export function View_MatTable_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),4,'mat-table',
      [['class','mat-table']],(null as any),(null as any),(null as any),View_MatTable_0,
      RenderType_MatTable)),i0.ɵdid(1,3391488,(null as any),3,i1.MatTable,[i0.IterableDiffers,
      i0.ChangeDetectorRef,i0.ElementRef,i0.Renderer2,[8,(null as any)]],(null as any),
      (null as any)),i0.ɵqud(603979776,1,{_columnDefs:1}),i0.ɵqud(335544320,2,{_headerDef:0}),
      i0.ɵqud(603979776,3,{_rowDefs:1})],(_ck,_v) => {
    _ck(_v,1,0);
  },(null as any));
}
export const MatTableNgFactory:i0.ComponentFactory<i1.MatTable<any>> = i0.ɵccf('mat-table',
    i1.MatTable,View_MatTable_Host_0,{trackBy:'trackBy',dataSource:'dataSource'},{},
    ([] as any[]));
const styles_MatHeaderRow:any[] = ([] as any[]);
export const RenderType_MatHeaderRow:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MatHeaderRow,
    data:{}});
export function View_MatHeaderRow_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵeld(0,16777216,(null as any),(null as any),1,(null as any),
      (null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(1,16384,(null as any),0,i5.CdkCellOutlet,[i0.ViewContainerRef],(null as any),
          (null as any))],(null as any),(null as any));
}
export function View_MatHeaderRow_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'mat-header-row',
      [['class','mat-header-row'],['role','row']],(null as any),(null as any),(null as any),
      View_MatHeaderRow_0,RenderType_MatHeaderRow)),i0.ɵdid(1,49152,(null as any),
      0,i1.MatHeaderRow,([] as any[]),(null as any),(null as any))],(null as any),
      (null as any));
}
export const MatHeaderRowNgFactory:i0.ComponentFactory<i1.MatHeaderRow> = i0.ɵccf('mat-header-row',
    i1.MatHeaderRow,View_MatHeaderRow_Host_0,{},{},([] as any[]));
const styles_MatRow:any[] = ([] as any[]);
export const RenderType_MatRow:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MatRow,
    data:{}});
export function View_MatRow_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵeld(0,16777216,(null as any),(null as any),1,(null as any),
      (null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(1,16384,(null as any),0,i5.CdkCellOutlet,[i0.ViewContainerRef],(null as any),
          (null as any))],(null as any),(null as any));
}
export function View_MatRow_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'mat-row',[['class',
      'mat-row'],['role','row']],(null as any),(null as any),(null as any),View_MatRow_0,
      RenderType_MatRow)),i0.ɵdid(1,49152,(null as any),0,i1.MatRow,([] as any[]),
      (null as any),(null as any))],(null as any),(null as any));
}
export const MatRowNgFactory:i0.ComponentFactory<i1.MatRow> = i0.ɵccf('mat-row',i1.MatRow,
    View_MatRow_Host_0,{},{},([] as any[]));
