// @angular/material/sort/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/material/sort';
import * as i2 from '@angular/common';
import * as i3 from '@angular/cdk/table';
export const MatSortModuleNgFactory:i0.NgModuleFactory<i1.MatSortModule> = i0.ɵcmf(i1.MatSortModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(4608,i1.MatSortHeaderIntl,
          i1.MatSortHeaderIntl,([] as any[])),i0.ɵmpd(512,i2.CommonModule,i2.CommonModule,
          ([] as any[])),i0.ɵmpd(512,i1.MatSortModule,i1.MatSortModule,([] as any[]))]);
    });
const styles_MatSortHeader:any[] = ['.mat-sort-header-container{display:flex;cursor:pointer}.mat-sort-header-position-before{flex-direction:row-reverse}.mat-sort-header-button{border:none;background:0 0;display:flex;align-items:center;padding:0;cursor:pointer;outline:0;font:inherit;color:currentColor}.mat-sort-header-arrow{height:12px;width:12px;margin:0 0 0 6px;position:relative;display:flex}.mat-sort-header-position-before .mat-sort-header-arrow{margin:0 6px 0 0}.mat-sort-header-stem{background:currentColor;height:10px;width:2px;margin:auto;display:flex;align-items:center}.mat-sort-header-indicator{width:100%;height:2px;display:flex;align-items:center;position:absolute;top:0;left:0;transition:225ms cubic-bezier(.4,0,.2,1)}.mat-sort-header-pointer-middle{margin:auto;height:2px;width:2px;background:currentColor;transform:rotate(45deg)}.mat-sort-header-pointer-left,.mat-sort-header-pointer-right{background:currentColor;width:6px;height:2px;transition:225ms cubic-bezier(.4,0,.2,1);position:absolute;top:0}.mat-sort-header-pointer-left{transform-origin:right;left:0}.mat-sort-header-pointer-right{transform-origin:left;right:0}'];
export const RenderType_MatSortHeader:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_MatSortHeader,data:{'animation':[{type:7,name:'indicator',definitions:[{type:0,
        name:'asc',styles:{type:6,styles:{transform:'translateY(0px)'},offset:(null as any)},
        options:(undefined as any)},{type:0,name:'desc',styles:{type:6,styles:{transform:'translateY(10px)'},
        offset:(null as any)},options:(undefined as any)},{type:1,expr:'asc <=> desc',
        animation:{type:4,styles:(null as any),timings:'225ms cubic-bezier(0.4,0.0,0.2,1)'},
        options:(null as any)}],options:{}},{type:7,name:'leftPointer',definitions:[{type:0,
        name:'asc',styles:{type:6,styles:{transform:'rotate(-45deg)'},offset:(null as any)},
        options:(undefined as any)},{type:0,name:'desc',styles:{type:6,styles:{transform:'rotate(45deg)'},
        offset:(null as any)},options:(undefined as any)},{type:1,expr:'asc <=> desc',
        animation:{type:4,styles:(null as any),timings:'225ms cubic-bezier(0.4,0.0,0.2,1)'},
        options:(null as any)}],options:{}},{type:7,name:'rightPointer',definitions:[{type:0,
        name:'asc',styles:{type:6,styles:{transform:'rotate(45deg)'},offset:(null as any)},
        options:(undefined as any)},{type:0,name:'desc',styles:{type:6,styles:{transform:'rotate(-45deg)'},
        offset:(null as any)},options:(undefined as any)},{type:1,expr:'asc <=> desc',
        animation:{type:4,styles:(null as any),timings:'225ms cubic-bezier(0.4,0.0,0.2,1)'},
        options:(null as any)}],options:{}},{type:7,name:'indicatorToggle',definitions:[{type:1,
        expr:'void => asc',animation:{type:4,styles:{type:5,steps:[{type:6,styles:{transform:'translateY(25%)',
            opacity:0},offset:(null as any)},{type:6,styles:{transform:'none',opacity:1},
            offset:(null as any)}]},timings:'225ms cubic-bezier(0.4,0.0,0.2,1)'},options:(null as any)},
        {type:1,expr:'asc => void',animation:{type:4,styles:{type:5,steps:[{type:6,
            styles:{transform:'none',opacity:1},offset:(null as any)},{type:6,styles:{transform:'translateY(-25%)',
            opacity:0},offset:(null as any)}]},timings:'225ms cubic-bezier(0.4,0.0,0.2,1)'},
            options:(null as any)},{type:1,expr:'void => desc',animation:{type:4,styles:{type:5,
            steps:[{type:6,styles:{transform:'translateY(-25%)',opacity:0},offset:(null as any)},
                {type:6,styles:{transform:'none',opacity:1},offset:(null as any)}]},
            timings:'225ms cubic-bezier(0.4,0.0,0.2,1)'},options:(null as any)},{type:1,
            expr:'desc => void',animation:{type:4,styles:{type:5,steps:[{type:6,styles:{transform:'none',
                opacity:1},offset:(null as any)},{type:6,styles:{transform:'translateY(25%)',
                opacity:0},offset:(null as any)}]},timings:'225ms cubic-bezier(0.4,0.0,0.2,1)'},
            options:(null as any)}],options:{}}]}});
function View_MatSortHeader_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),5,'div',[['class',
      'mat-sort-header-arrow']],[[24,'@indicatorToggle',0]],(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵeld(1,0,(null as any),(null as any),
      0,'div',[['class','mat-sort-header-stem']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵeld(2,0,(null as any),(null as any),
      3,'div',[['class','mat-sort-header-indicator']],[[24,'@indicator',0]],(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵeld(3,0,(null as any),
      (null as any),0,'div',[['class','mat-sort-header-pointer-left']],[[24,'@leftPointer',
          0]],(null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵeld(4,
      0,(null as any),(null as any),0,'div',[['class','mat-sort-header-pointer-right']],
      [[24,'@rightPointer',0]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵeld(5,0,(null as any),(null as any),0,'div',[['class','mat-sort-header-pointer-middle']],
          (null as any),(null as any),(null as any),(null as any),(null as any)))],
      (null as any),(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co._sort.direction;
        _ck(_v,0,0,currVal_0);
        const currVal_1:any = _co._sort.direction;
        _ck(_v,2,0,currVal_1);
        const currVal_2:any = _co._sort.direction;
        _ck(_v,3,0,currVal_2);
        const currVal_3:any = _co._sort.direction;
        _ck(_v,4,0,currVal_3);
      });
}
function View_MatSortHeader_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'span',[['class',
      'cdk-visually-hidden']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted(1,(null as any),['','']))],(null as any),(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co._intl.sortDescriptionLabel(_co.id,_co._sort.direction);
    _ck(_v,1,0,currVal_0);
  });
}
export function View_MatSortHeader_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),4,'div',[['class',
      'mat-sort-header-container']],[[2,'mat-sort-header-position-before',(null as any)]],
      (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵeld(1,
      0,(null as any),(null as any),1,'button',[['class','mat-sort-header-button'],
          ['type','button']],[[1,'aria-label',0]],(null as any),(null as any),(null as any),
      (null as any))),i0.ɵncd((null as any),0),(_l()(),i0.ɵand(16777216,(null as any),
      (null as any),1,(null as any),View_MatSortHeader_1)),i0.ɵdid(4,16384,(null as any),
      0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MatSortHeader_2)),
      i0.ɵdid(6,16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],
          {ngIf:[0,'ngIf']},(null as any))],(_ck,_v) => {
    var _co:i1.MatSortHeader = _v.component;
    const currVal_2:any = _co._isSorted();
    _ck(_v,4,0,currVal_2);
    const currVal_3:any = _co._isSorted();
    _ck(_v,6,0,currVal_3);
  },(_ck,_v) => {
    var _co:i1.MatSortHeader = _v.component;
    const currVal_0:any = (_co.arrowPosition == 'before');
    _ck(_v,0,0,currVal_0);
    const currVal_1:any = _co._intl.sortButtonLabel(_co.id);
    _ck(_v,1,0,currVal_1);
  });
}
export function View_MatSortHeader_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'div',[['mat-sort-header',
      '']],[[2,'mat-sort-header-sorted',(null as any)]],[[(null as any),'click']],
      (_v,en,$event) => {
        var ad:boolean = true;
        if (('click' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,1)._sort.sort(i0.ɵnov(_v,1))) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },View_MatSortHeader_0,RenderType_MatSortHeader)),i0.ɵdid(1,245760,(null as any),
      0,i1.MatSortHeader,[i1.MatSortHeaderIntl,i0.ChangeDetectorRef,[2,i1.MatSort],
          [2,i3.CdkColumnDef]],{id:[0,'id']},(null as any))],(_ck,_v) => {
    const currVal_1:any = '';
    _ck(_v,1,0,currVal_1);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,1)._isSorted();
    _ck(_v,0,0,currVal_0);
  });
}
export const MatSortHeaderNgFactory:i0.ComponentFactory<i1.MatSortHeader> = i0.ɵccf('[mat-sort-header]',
    i1.MatSortHeader,View_MatSortHeader_Host_0,{id:'mat-sort-header',arrowPosition:'arrowPosition',
        start:'start',disableClear:'disableClear'},{},['*']);
