// @angular/material/dialog/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/material/dialog';
import * as i2 from '@angular/common';
import * as i3 from '@angular/cdk/platform';
import * as i4 from '@angular/cdk/scrolling';
import * as i5 from '@angular/cdk/overlay';
import * as i6 from '@angular/cdk/a11y';
import * as i7 from '@angular/cdk/bidi';
import * as i8 from '@angular/platform-browser';
import * as i9 from '@angular/cdk/portal';
import * as i10 from '@angular/material/core';
export const MatDialogModuleNgFactory:i0.NgModuleFactory<i1.MatDialogModule> = i0.ɵcmf(i1.MatDialogModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,[MatDialogContainerNgFactory]],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),
          i0.ɵmpd(4608,i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(4608,
              i3.Platform,i3.Platform,([] as any[])),i0.ɵmpd(5120,i4.ScrollDispatcher,
              i4.SCROLL_DISPATCHER_PROVIDER_FACTORY,[[3,i4.ScrollDispatcher],i0.NgZone,
                  i3.Platform]),i0.ɵmpd(5120,i4.ViewportRuler,i4.VIEWPORT_RULER_PROVIDER_FACTORY,
              [[3,i4.ViewportRuler],i3.Platform,i0.NgZone,i4.ScrollDispatcher]),i0.ɵmpd(4608,
              i5.ScrollStrategyOptions,i5.ScrollStrategyOptions,[i4.ScrollDispatcher,
                  i4.ViewportRuler]),i0.ɵmpd(5120,i5.OverlayContainer,i5.ɵa,[[3,i5.OverlayContainer]]),
          i0.ɵmpd(4608,i5.ɵf,i5.ɵf,[i4.ViewportRuler]),i0.ɵmpd(4608,i5.Overlay,i5.Overlay,
              [i5.ScrollStrategyOptions,i5.OverlayContainer,i0.ComponentFactoryResolver,
                  i5.ɵf,i0.ApplicationRef,i0.Injector,i0.NgZone]),i0.ɵmpd(5120,i5.ɵc,
              i5.ɵd,[i5.Overlay]),i0.ɵmpd(4608,i6.InteractivityChecker,i6.InteractivityChecker,
              [i3.Platform]),i0.ɵmpd(4608,i6.FocusTrapFactory,i6.FocusTrapFactory,
              [i6.InteractivityChecker,i3.Platform,i0.NgZone]),i0.ɵmpd(136192,i6.AriaDescriber,
              i6.ARIA_DESCRIBER_PROVIDER_FACTORY,[[3,i6.AriaDescriber],i3.Platform]),
          i0.ɵmpd(5120,i6.LiveAnnouncer,i6.LIVE_ANNOUNCER_PROVIDER_FACTORY,[[3,i6.LiveAnnouncer],
              [2,i6.LIVE_ANNOUNCER_ELEMENT_TOKEN],i3.Platform]),i0.ɵmpd(5120,i6.FocusMonitor,
              i6.FOCUS_MONITOR_PROVIDER_FACTORY,[[3,i6.FocusMonitor],i0.NgZone,i3.Platform]),
          i0.ɵmpd(6144,i7.DIR_DOCUMENT,(null as any),[i8.DOCUMENT]),i0.ɵmpd(4608,i7.Directionality,
              i7.Directionality,[[2,i7.DIR_DOCUMENT]]),i0.ɵmpd(5120,i1.MAT_DIALOG_SCROLL_STRATEGY,
              i1.MAT_DIALOG_SCROLL_STRATEGY_PROVIDER_FACTORY,[i5.Overlay]),i0.ɵmpd(4608,
              i1.MatDialog,i1.MatDialog,[i5.Overlay,i0.Injector,[2,i2.Location],i1.MAT_DIALOG_SCROLL_STRATEGY,
                  [3,i1.MatDialog]]),i0.ɵmpd(512,i2.CommonModule,i2.CommonModule,([] as any[])),
          i0.ɵmpd(512,i9.PortalModule,i9.PortalModule,([] as any[])),i0.ɵmpd(512,i3.PlatformModule,
              i3.PlatformModule,([] as any[])),i0.ɵmpd(512,i4.ScrollDispatchModule,
              i4.ScrollDispatchModule,([] as any[])),i0.ɵmpd(512,i5.OverlayModule,
              i5.OverlayModule,([] as any[])),i0.ɵmpd(512,i6.A11yModule,i6.A11yModule,
              ([] as any[])),i0.ɵmpd(512,i10.CompatibilityModule,i10.CompatibilityModule,
              ([] as any[])),i0.ɵmpd(512,i7.BidiModule,i7.BidiModule,([] as any[])),
          i0.ɵmpd(256,i10.MATERIAL_SANITY_CHECKS,true,([] as any[])),i0.ɵmpd(512,i10.MatCommonModule,
              i10.MatCommonModule,[[2,i10.MATERIAL_SANITY_CHECKS]]),i0.ɵmpd(512,i1.MatDialogModule,
              i1.MatDialogModule,([] as any[]))]);
    });
const styles_MatDialogContainer:any[] = ['.mat-dialog-container{box-shadow:0 11px 15px -7px rgba(0,0,0,.2),0 24px 38px 3px rgba(0,0,0,.14),0 9px 46px 8px rgba(0,0,0,.12);display:block;padding:24px;border-radius:2px;box-sizing:border-box;overflow:auto;max-width:80vw;outline:0;width:100%;height:100%}@media screen and (-ms-high-contrast:active){.mat-dialog-container{outline:solid 1px}}.mat-dialog-content{display:block;margin:0 -24px;padding:0 24px;max-height:65vh;overflow:auto;-webkit-overflow-scrolling:touch;-webkit-backface-visibility:hidden;backface-visibility:hidden}.mat-dialog-title{margin:0 0 20px;display:block}.mat-dialog-actions{padding:12px 0;display:flex;flex-wrap:wrap}.mat-dialog-actions:last-child{margin-bottom:-24px}.mat-dialog-actions[align=end]{justify-content:flex-end}.mat-dialog-actions[align=center]{justify-content:center}.mat-dialog-actions .mat-button+.mat-button,.mat-dialog-actions .mat-button+.mat-raised-button,.mat-dialog-actions .mat-raised-button+.mat-button,.mat-dialog-actions .mat-raised-button+.mat-raised-button{margin-left:8px}[dir=rtl] .mat-dialog-actions .mat-button+.mat-button,[dir=rtl] .mat-dialog-actions .mat-button+.mat-raised-button,[dir=rtl] .mat-dialog-actions .mat-raised-button+.mat-button,[dir=rtl] .mat-dialog-actions .mat-raised-button+.mat-raised-button{margin-left:0;margin-right:8px}'];
export const RenderType_MatDialogContainer:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_MatDialogContainer,data:{'animation':[{type:7,name:'slideDialog',
        definitions:[{type:0,name:'enter',styles:{type:6,styles:{transform:'none',
            opacity:1},offset:(null as any)},options:(undefined as any)},{type:0,name:'void',
            styles:{type:6,styles:{transform:'translate3d(0, 25%, 0) scale(0.9)',opacity:0},
                offset:(null as any)},options:(undefined as any)},{type:0,name:'exit',
            styles:{type:6,styles:{transform:'translate3d(0, 25%, 0)',opacity:0},offset:(null as any)},
            options:(undefined as any)},{type:1,expr:'* => *',animation:{type:4,styles:(null as any),
            timings:'400ms cubic-bezier(0.25, 0.8, 0.25, 1)'},options:(null as any)}],
        options:{}}]}});
function View_MatDialogContainer_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(0,(null as any),(null as any),0))],(null as any),
      (null as any));
}
export function View_MatDialogContainer_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵqud(402653184,1,{_portalHost:0}),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_MatDialogContainer_1)),i0.ɵdid(2,
      147456,[[1,4]],0,i9.PortalHostDirective,[i0.ComponentFactoryResolver,i0.ViewContainerRef],
      {portal:[0,'portal']},(null as any))],(_ck,_v) => {
    const currVal_0:any = '';
    _ck(_v,2,0,currVal_0);
  },(null as any));
}
export function View_MatDialogContainer_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'mat-dialog-container',
      [['class','mat-dialog-container'],['tabindex','-1']],[[1,'role',0],[1,'aria-labelledby',
          0],[1,'aria-describedby',0],[40,'@slideDialog',0]],[['component','@slideDialog.start'],
          ['component','@slideDialog.done']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('component:@slideDialog.start' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,1)._onAnimationStart($event)) !== false);
          ad = (pd_0 && ad);
        }
        if (('component:@slideDialog.done' === en)) {
          const pd_1:any = ((<any>i0.ɵnov(_v,1)._onAnimationDone($event)) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },View_MatDialogContainer_0,RenderType_MatDialogContainer)),i0.ɵdid(1,49152,
      (null as any),0,i1.MatDialogContainer,[i0.ElementRef,i6.FocusTrapFactory,i0.ChangeDetectorRef,
          [2,i8.DOCUMENT]],(null as any),(null as any))],(null as any),(_ck,_v) => {
    const currVal_0:any = ((i0.ɵnov(_v,1)._config == null)? (null as any): i0.ɵnov(_v,
        1)._config.role);
    const currVal_1:any = i0.ɵnov(_v,1)._ariaLabelledBy;
    const currVal_2:any = (((i0.ɵnov(_v,1)._config == null)? (null as any): i0.ɵnov(_v,
        1)._config.ariaDescribedBy) || (null as any));
    const currVal_3:any = i0.ɵnov(_v,1)._state;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
export const MatDialogContainerNgFactory:i0.ComponentFactory<i1.MatDialogContainer> = i0.ɵccf('mat-dialog-container',
    i1.MatDialogContainer,View_MatDialogContainer_Host_0,{},{},([] as any[]));
