// @angular/material/toolbar/typings/index.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from '@angular/material/toolbar';
import * as i2 from '@angular/cdk/bidi';
import * as i3 from '@angular/platform-browser';
import * as i4 from '@angular/material/core';
export var MatToolbarModuleNgFactory = i0.ɵcmf(i1.MatToolbarModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(6144, i2.DIR_DOCUMENT, null, [i3.DOCUMENT]), i0.ɵmpd(4608, i2.Directionality, i2.Directionality, [[2, i2.DIR_DOCUMENT]]), i0.ɵmpd(512, i4.CompatibilityModule, i4.CompatibilityModule, []), i0.ɵmpd(512, i2.BidiModule, i2.BidiModule, []), i0.ɵmpd(256, i4.MATERIAL_SANITY_CHECKS, true, []),
        i0.ɵmpd(512, i4.MatCommonModule, i4.MatCommonModule, [[2, i4.MATERIAL_SANITY_CHECKS]]),
        i0.ɵmpd(512, i1.MatToolbarModule, i1.MatToolbarModule, [])]);
});
var styles_MatToolbar = ['.mat-toolbar{display:flex;box-sizing:border-box;width:100%;padding:0 16px;flex-direction:column}.mat-toolbar .mat-toolbar-row{display:flex;box-sizing:border-box;width:100%;flex-direction:row;align-items:center;white-space:nowrap}.mat-toolbar{min-height:64px}.mat-toolbar-row{height:64px}@media (max-width:600px){.mat-toolbar{min-height:56px}.mat-toolbar-row{height:56px}}'];
export var RenderType_MatToolbar = i0.ɵcrt({ encapsulation: 2, styles: styles_MatToolbar,
    data: {} });
export function View_MatToolbar_0(_l) {
    return i0.ɵvid(2, [(_l()(), i0.ɵeld(0, 0, null, null, 4, 'div', [['class',
                'mat-toolbar-layout']], null, null, null, null, null)), (_l()(), i0.ɵeld(1, 0, null, null, 2, 'mat-toolbar-row', [['class', 'mat-toolbar-row']], null, null, null, null, null)), i0.ɵdid(2, 16384, null, 0, i1.MatToolbarRow, [], null, null), i0.ɵncd(null, 0), i0.ɵncd(null, 1)], null, null);
}
export function View_MatToolbar_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'mat-toolbar', [['class', 'mat-toolbar'], ['role', 'toolbar']], null, null, null, View_MatToolbar_0, RenderType_MatToolbar)), i0.ɵdid(1, 49152, null, 0, i1.MatToolbar, [i0.Renderer2, i0.ElementRef], null, null)], null, null);
}
export var MatToolbarNgFactory = i0.ɵccf('mat-toolbar', i1.MatToolbar, View_MatToolbar_Host_0, { color: 'color' }, {}, ['*', 'mat-toolbar-row']);
