// @angular/material/select/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/material/select';
import * as i2 from '@angular/common';
import * as i3 from '@angular/cdk/platform';
import * as i4 from '@angular/cdk/scrolling';
import * as i5 from '@angular/cdk/overlay';
import * as i6 from '@angular/cdk/bidi';
import * as i7 from '@angular/platform-browser';
import * as i8 from '@angular/material/core';
import * as i9 from '@angular/cdk/portal';
import * as i10 from '@angular/material/form-field';
import * as i11 from '@angular/forms';
export const MatSelectModuleNgFactory:i0.NgModuleFactory<i1.MatSelectModule> = i0.ɵcmf(i1.MatSelectModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(4608,i3.Platform,
          i3.Platform,([] as any[])),i0.ɵmpd(5120,i4.ScrollDispatcher,i4.SCROLL_DISPATCHER_PROVIDER_FACTORY,
          [[3,i4.ScrollDispatcher],i0.NgZone,i3.Platform]),i0.ɵmpd(5120,i4.ViewportRuler,
          i4.VIEWPORT_RULER_PROVIDER_FACTORY,[[3,i4.ViewportRuler],i3.Platform,i0.NgZone,
              i4.ScrollDispatcher]),i0.ɵmpd(4608,i5.ScrollStrategyOptions,i5.ScrollStrategyOptions,
          [i4.ScrollDispatcher,i4.ViewportRuler]),i0.ɵmpd(5120,i5.OverlayContainer,
          i5.ɵa,[[3,i5.OverlayContainer]]),i0.ɵmpd(4608,i5.ɵf,i5.ɵf,[i4.ViewportRuler]),
          i0.ɵmpd(4608,i5.Overlay,i5.Overlay,[i5.ScrollStrategyOptions,i5.OverlayContainer,
              i0.ComponentFactoryResolver,i5.ɵf,i0.ApplicationRef,i0.Injector,i0.NgZone]),
          i0.ɵmpd(5120,i5.ɵc,i5.ɵd,[i5.Overlay]),i0.ɵmpd(6144,i6.DIR_DOCUMENT,(null as any),
              [i7.DOCUMENT]),i0.ɵmpd(4608,i6.Directionality,i6.Directionality,[[2,
              i6.DIR_DOCUMENT]]),i0.ɵmpd(5120,i1.MAT_SELECT_SCROLL_STRATEGY,i1.MAT_SELECT_SCROLL_STRATEGY_PROVIDER_FACTORY,
              [i5.Overlay]),i0.ɵmpd(4608,i8.ErrorStateMatcher,i8.ErrorStateMatcher,
              ([] as any[])),i0.ɵmpd(512,i2.CommonModule,i2.CommonModule,([] as any[])),
          i0.ɵmpd(512,i9.PortalModule,i9.PortalModule,([] as any[])),i0.ɵmpd(512,i3.PlatformModule,
              i3.PlatformModule,([] as any[])),i0.ɵmpd(512,i4.ScrollDispatchModule,
              i4.ScrollDispatchModule,([] as any[])),i0.ɵmpd(512,i5.OverlayModule,
              i5.OverlayModule,([] as any[])),i0.ɵmpd(512,i8.CompatibilityModule,i8.CompatibilityModule,
              ([] as any[])),i0.ɵmpd(512,i6.BidiModule,i6.BidiModule,([] as any[])),
          i0.ɵmpd(256,i8.MATERIAL_SANITY_CHECKS,true,([] as any[])),i0.ɵmpd(512,i8.MatCommonModule,
              i8.MatCommonModule,[[2,i8.MATERIAL_SANITY_CHECKS]]),i0.ɵmpd(512,i8.MatRippleModule,
              i8.MatRippleModule,([] as any[])),i0.ɵmpd(512,i8.MatPseudoCheckboxModule,
              i8.MatPseudoCheckboxModule,([] as any[])),i0.ɵmpd(512,i8.MatOptionModule,
              i8.MatOptionModule,([] as any[])),i0.ɵmpd(512,i10.MatFormFieldModule,
              i10.MatFormFieldModule,([] as any[])),i0.ɵmpd(512,i1.MatSelectModule,
              i1.MatSelectModule,([] as any[]))]);
    });
const styles_MatSelect:any[] = ['.mat-select{display:inline-block;width:100%;outline:0}.mat-select-trigger{display:inline-table;cursor:pointer;position:relative;box-sizing:border-box}.mat-select-disabled .mat-select-trigger{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;cursor:default}.mat-select-value{display:table-cell;max-width:0;width:100%;overflow:hidden;text-overflow:ellipsis}.mat-select-value-text{white-space:nowrap;overflow:hidden;text-overflow:ellipsis}.mat-select-arrow-wrapper{display:table-cell;vertical-align:middle}.mat-select-arrow{width:0;height:0;border-left:5px solid transparent;border-right:5px solid transparent;border-top:5px solid;margin:0 4px}.mat-select-panel{min-width:112px;max-width:280px;overflow:auto;-webkit-overflow-scrolling:touch;padding-top:0;padding-bottom:0;max-height:256px;min-width:100%}.mat-select-panel:not([class*=mat-elevation-z]){box-shadow:0 5px 5px -3px rgba(0,0,0,.2),0 8px 10px 1px rgba(0,0,0,.14),0 3px 14px 2px rgba(0,0,0,.12)}@media screen and (-ms-high-contrast:active){.mat-select-panel{outline:solid 1px}}.mat-select-panel .mat-optgroup-label,.mat-select-panel .mat-option{font-size:inherit;line-height:3em;height:3em}.mat-form-field-type-mat-select .mat-form-field-flex{cursor:pointer}.mat-form-field-type-mat-select .mat-form-field-placeholder{width:calc(100% - 18px)}'];
export const RenderType_MatSelect:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MatSelect,
    data:{'animation':[{type:7,name:'transformPanel',definitions:[{type:0,name:'showing',
        styles:{type:6,styles:{opacity:1,minWidth:'calc(100% + 32px)',transform:'scaleY(1)'},
            offset:(null as any)},options:(undefined as any)},{type:0,name:'showing-multiple',
        styles:{type:6,styles:{opacity:1,minWidth:'calc(100% + 64px)',transform:'scaleY(1)'},
            offset:(null as any)},options:(undefined as any)},{type:1,expr:'void => *',
        animation:[{type:6,styles:{opacity:0,minWidth:'100%',transform:'scaleY(0)'},
            offset:(null as any)},{type:4,styles:(null as any),timings:'150ms cubic-bezier(0.25, 0.8, 0.25, 1)'}],
        options:(null as any)},{type:1,expr:'* => void',animation:[{type:4,styles:{type:6,
        styles:{opacity:0},offset:(null as any)},timings:'250ms 100ms linear'}],options:(null as any)}],
        options:{}},{type:7,name:'fadeInContent',definitions:[{type:0,name:'showing',
        styles:{type:6,styles:{opacity:1},offset:(null as any)},options:(undefined as any)},
        {type:1,expr:'void => showing',animation:[{type:6,styles:{opacity:0},offset:(null as any)},
            {type:4,styles:(null as any),timings:'150ms 100ms cubic-bezier(0.55, 0, 0.55, 0.2)'}],
            options:(null as any)}],options:{}}]}});
function View_MatSelect_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,(null as any),
      (null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted(-1,(null as any),[' ']))],(null as any),(null as any));
}
function View_MatSelect_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'span',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted(1,(null as any),['','']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.triggerValue;
    _ck(_v,1,0,currVal_0);
  });
}
function View_MatSelect_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵncd((null as any),0),(_l()(),i0.ɵand(0,(null as any),(null as any),
      0))],(null as any),(null as any));
}
function View_MatSelect_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),5,'span',[['class',
      'mat-select-value-text']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),i0.ɵdid(1,16384,(null as any),0,i2.NgSwitch,([] as any[]),{ngSwitch:[0,
      'ngSwitch']},(null as any)),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
      1,(null as any),View_MatSelect_3)),i0.ɵdid(3,16384,(null as any),0,i2.NgSwitchDefault,
      [i0.ViewContainerRef,i0.TemplateRef,i2.NgSwitch],(null as any),(null as any)),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MatSelect_4)),
      i0.ɵdid(5,278528,(null as any),0,i2.NgSwitchCase,[i0.ViewContainerRef,i0.TemplateRef,
          i2.NgSwitch],{ngSwitchCase:[0,'ngSwitchCase']},(null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:boolean = !!_co.customTrigger;
    _ck(_v,1,0,currVal_0);
    const currVal_1:any = true;
    _ck(_v,5,0,currVal_1);
  },(null as any));
}
function View_MatSelect_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,[[2,0],['panel',1]],(null as any),3,'div',
      ([] as any[]),[[24,'@transformPanel',0],[4,'transformOrigin',(null as any)],
          [2,'mat-select-panel-done-animating',(null as any)],[4,'font-size','px']],
      [[(null as any),'@transformPanel.done']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('@transformPanel.done' === en)) {
          const pd_0:any = ((<any>_co._onPanelDone()) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(1,278528,(null as any),0,i2.NgClass,
      [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
          'klass'],ngClass:[1,'ngClass']},(null as any)),(_l()(),i0.ɵeld(2,0,(null as any),
      (null as any),1,'div',[['class','mat-select-content']],[[24,'@fadeInContent',
          0]],[[(null as any),'@fadeInContent.done']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('@fadeInContent.done' === en)) {
          const pd_0:any = ((<any>_co._onFadeInDone()) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵncd((null as any),1)],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_4:any = i0.ɵinlineInterpolate(1,'mat-select-panel ',_co._getPanelTheme(),
        '');
    const currVal_5:any = _co.panelClass;
    _ck(_v,1,0,currVal_4,currVal_5);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = (_co.multiple? 'showing-multiple': 'showing');
    const currVal_1:any = _co._transformOrigin;
    const currVal_2:any = _co._panelDoneAnimating;
    const currVal_3:any = _co._triggerFontSize;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
    const currVal_6:any = 'showing';
    _ck(_v,2,0,currVal_6);
  });
}
export function View_MatSelect_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(402653184,1,{trigger:0}),i0.ɵqud(671088640,2,{panel:0}),
      i0.ɵqud(402653184,3,{overlayDir:0}),(_l()(),i0.ɵeld(3,0,[[1,0],['trigger',1]],
          (null as any),8,'div',[['aria-hidden','true'],['cdk-overlay-origin',''],
              ['class','mat-select-trigger']],(null as any),[[(null as any),'click']],
          (_v,en,$event) => {
            var ad:boolean = true;
            var _co:i1.MatSelect = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.toggle()) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(4,16384,[['origin',4]],0,i5.OverlayOrigin,
          [i0.ElementRef],(null as any),(null as any)),(_l()(),i0.ɵeld(5,0,(null as any),
          (null as any),4,'div',[['class','mat-select-value']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_MatSelect_1)),i0.ɵdid(7,16384,(null as any),
          0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MatSelect_2)),
      i0.ɵdid(9,16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],
          {ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵeld(10,0,(null as any),(null as any),
          1,'div',[['class','mat-select-arrow-wrapper']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵeld(11,0,(null as any),
          (null as any),0,'div',[['class','mat-select-arrow']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(_v,en,$event) => {
            var ad:boolean = true;
            var _co:i1.MatSelect = _v.component;
            if (('backdropClick' === en)) {
              const pd_0:any = ((<any>_co.close()) !== false);
              ad = (pd_0 && ad);
            }
            if (('attach' === en)) {
              const pd_1:any = ((<any>_co._onAttached()) !== false);
              ad = (pd_1 && ad);
            }
            if (('detach' === en)) {
              const pd_2:any = ((<any>_co.close()) !== false);
              ad = (pd_2 && ad);
            }
            return ad;
          },View_MatSelect_5)),i0.ɵdid(13,671744,[[3,4]],0,i5.ConnectedOverlayDirective,
          [i5.Overlay,i0.Renderer2,i0.TemplateRef,i0.ViewContainerRef,i5.ɵc,[2,i6.Directionality]],
          {_deprecatedOrigin:[0,'_deprecatedOrigin'],_deprecatedPositions:[1,'_deprecatedPositions'],
              _deprecatedOffsetY:[2,'_deprecatedOffsetY'],_deprecatedMinWidth:[3,'_deprecatedMinWidth'],
              _deprecatedBackdropClass:[4,'_deprecatedBackdropClass'],_deprecatedScrollStrategy:[5,
                  '_deprecatedScrollStrategy'],_deprecatedOpen:[6,'_deprecatedOpen'],
              _deprecatedHasBackdrop:[7,'_deprecatedHasBackdrop']},{backdropClick:'backdropClick',
              attach:'attach',detach:'detach'})],(_ck,_v) => {
    var _co:i1.MatSelect = _v.component;
    const currVal_0:any = _co.empty;
    _ck(_v,7,0,currVal_0);
    const currVal_1:boolean = !_co.empty;
    _ck(_v,9,0,currVal_1);
    const currVal_2:any = i0.ɵnov(_v,4);
    const currVal_3:any = _co._positions;
    const currVal_4:any = _co._offsetY;
    const currVal_5:any = ((_co._triggerRect == null)? (null as any): _co._triggerRect.width);
    const currVal_6:any = 'cdk-overlay-transparent-backdrop';
    const currVal_7:any = _co._scrollStrategy;
    const currVal_8:any = _co.panelOpen;
    const currVal_9:any = '';
    _ck(_v,13,0,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8,
        currVal_9);
  },(null as any));
}
export function View_MatSelect_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),5,'mat-select',
      [['class','mat-select'],['role','listbox']],[[1,'id',0],[1,'tabindex',0],[1,
          'aria-label',0],[1,'aria-labelledby',0],[1,'aria-required',0],[1,'aria-disabled',
          0],[1,'aria-invalid',0],[1,'aria-owns',0],[1,'aria-multiselectable',0],[1,
          'aria-describedby',0],[1,'aria-activedescendant',0],[2,'mat-select-disabled',
          (null as any)],[2,'mat-select-invalid',(null as any)],[2,'mat-select-required',
          (null as any)]],[[(null as any),'keydown'],[(null as any),'focus'],[(null as any),
          'blur']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('keydown' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,2)._handleKeydown($event)) !== false);
          ad = (pd_0 && ad);
        }
        if (('focus' === en)) {
          const pd_1:any = ((<any>i0.ɵnov(_v,2)._onFocus()) !== false);
          ad = (pd_1 && ad);
        }
        if (('blur' === en)) {
          const pd_2:any = ((<any>i0.ɵnov(_v,2)._onBlur()) !== false);
          ad = (pd_2 && ad);
        }
        return ad;
      },View_MatSelect_0,RenderType_MatSelect)),i0.ɵprd(6144,(null as any),i10.MatFormFieldControl,
      (null as any),[i1.MatSelect]),i0.ɵdid(2,1294336,(null as any),3,i1.MatSelect,
      [i4.ViewportRuler,i0.ChangeDetectorRef,i0.NgZone,i8.ErrorStateMatcher,i0.Renderer2,
          i0.ElementRef,[2,i6.Directionality],[2,i11.NgForm],[2,i11.FormGroupDirective],
          [2,i10.MatFormField],[8,(null as any)],[8,(null as any)],i1.MAT_SELECT_SCROLL_STRATEGY],
      (null as any),(null as any)),i0.ɵqud(603979776,1,{options:1}),i0.ɵqud(603979776,
      2,{optionGroups:1}),i0.ɵqud(335544320,3,{customTrigger:0})],(_ck,_v) => {
    _ck(_v,2,0);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,2).id;
    const currVal_1:any = i0.ɵnov(_v,2).tabIndex;
    const currVal_2:any = i0.ɵnov(_v,2)._ariaLabel;
    const currVal_3:any = i0.ɵnov(_v,2).ariaLabelledby;
    const currVal_4:any = i0.ɵnov(_v,2).required.toString();
    const currVal_5:any = i0.ɵnov(_v,2).disabled.toString();
    const currVal_6:any = i0.ɵnov(_v,2).errorState;
    const currVal_7:any = i0.ɵnov(_v,2)._optionIds;
    const currVal_8:any = i0.ɵnov(_v,2).multiple;
    const currVal_9:any = (i0.ɵnov(_v,2)._ariaDescribedby || (null as any));
    const currVal_10:any = i0.ɵnov(_v,2)._getAriaActiveDescendant();
    const currVal_11:any = i0.ɵnov(_v,2).disabled;
    const currVal_12:any = i0.ɵnov(_v,2).errorState;
    const currVal_13:any = i0.ɵnov(_v,2).required;
    _ck(_v,0,1,[currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,
        currVal_7,currVal_8,currVal_9,currVal_10,currVal_11,currVal_12,currVal_13]);
  });
}
export const MatSelectNgFactory:i0.ComponentFactory<i1.MatSelect> = i0.ɵccf('mat-select',
    i1.MatSelect,View_MatSelect_Host_0,{disabled:'disabled',tabIndex:'tabIndex',panelClass:'panelClass',
        placeholder:'placeholder',required:'required',multiple:'multiple',compareWith:'compareWith',
        value:'value',disableRipple:'disableRipple',ariaLabel:'aria-label',ariaLabelledby:'aria-labelledby',
        errorStateMatcher:'errorStateMatcher',id:'id'},{onOpen:'onOpen',onClose:'onClose',
        change:'change',valueChange:'valueChange'},['mat-select-trigger','*']);
