// @angular/material/grid-list/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/material/grid-list';
import * as i2 from '@angular/cdk/bidi';
import * as i3 from '@angular/platform-browser';
import * as i4 from '@angular/material/core';
export const MatGridListModuleNgFactory:i0.NgModuleFactory<i1.MatGridListModule> = i0.ɵcmf(i1.MatGridListModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(6144,
          i2.DIR_DOCUMENT,(null as any),[i3.DOCUMENT]),i0.ɵmpd(4608,i2.Directionality,
          i2.Directionality,[[2,i2.DIR_DOCUMENT]]),i0.ɵmpd(512,i4.CompatibilityModule,
          i4.CompatibilityModule,([] as any[])),i0.ɵmpd(512,i2.BidiModule,i2.BidiModule,
          ([] as any[])),i0.ɵmpd(256,i4.MATERIAL_SANITY_CHECKS,true,([] as any[])),
          i0.ɵmpd(512,i4.MatCommonModule,i4.MatCommonModule,[[2,i4.MATERIAL_SANITY_CHECKS]]),
          i0.ɵmpd(512,i4.MatLineModule,i4.MatLineModule,([] as any[])),i0.ɵmpd(512,
              i1.MatGridListModule,i1.MatGridListModule,([] as any[]))]);
    });
const styles_MatGridList:any[] = ['.mat-grid-list{display:block;position:relative}.mat-grid-tile{display:block;position:absolute;overflow:hidden}.mat-grid-tile .mat-figure{top:0;left:0;right:0;bottom:0;position:absolute;display:flex;align-items:center;justify-content:center;height:100%;padding:0;margin:0}.mat-grid-tile .mat-grid-tile-footer,.mat-grid-tile .mat-grid-tile-header{display:flex;align-items:center;height:48px;color:#fff;background:rgba(0,0,0,.38);overflow:hidden;padding:0 16px;position:absolute;left:0;right:0}.mat-grid-tile .mat-grid-tile-footer>*,.mat-grid-tile .mat-grid-tile-header>*{margin:0;padding:0;font-weight:400;font-size:inherit}.mat-grid-tile .mat-grid-tile-footer.mat-2-line,.mat-grid-tile .mat-grid-tile-header.mat-2-line{height:68px}.mat-grid-tile .mat-grid-list-text{display:flex;flex-direction:column;width:100%;box-sizing:border-box;overflow:hidden}.mat-grid-tile .mat-grid-list-text>*{margin:0;padding:0;font-weight:400;font-size:inherit}.mat-grid-tile .mat-grid-list-text:empty{display:none}.mat-grid-tile .mat-grid-tile-header{top:0}.mat-grid-tile .mat-grid-tile-footer{bottom:0}.mat-grid-tile .mat-grid-avatar{padding-right:16px}[dir=rtl] .mat-grid-tile .mat-grid-avatar{padding-right:0;padding-left:16px}.mat-grid-tile .mat-grid-avatar:empty{display:none}'];
export const RenderType_MatGridList:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MatGridList,
    data:{}});
export function View_MatGridList_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'div',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),i0.ɵncd((null as any),
      0)],(null as any),(null as any));
}
export function View_MatGridList_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),2,'mat-grid-list',
      [['class','mat-grid-list']],(null as any),(null as any),(null as any),View_MatGridList_0,
      RenderType_MatGridList)),i0.ɵdid(1,2211840,(null as any),1,i1.MatGridList,[i0.Renderer2,
      i0.ElementRef,[2,i2.Directionality]],(null as any),(null as any)),i0.ɵqud(603979776,
      1,{_tiles:1})],(_ck,_v) => {
    _ck(_v,1,0);
  },(null as any));
}
export const MatGridListNgFactory:i0.ComponentFactory<i1.MatGridList> = i0.ɵccf('mat-grid-list',
    i1.MatGridList,View_MatGridList_Host_0,{cols:'cols',gutterSize:'gutterSize',rowHeight:'rowHeight'},
    {},['*']);
const styles_MatGridTile:any[] = ['.mat-grid-list{display:block;position:relative}.mat-grid-tile{display:block;position:absolute;overflow:hidden}.mat-grid-tile .mat-figure{top:0;left:0;right:0;bottom:0;position:absolute;display:flex;align-items:center;justify-content:center;height:100%;padding:0;margin:0}.mat-grid-tile .mat-grid-tile-footer,.mat-grid-tile .mat-grid-tile-header{display:flex;align-items:center;height:48px;color:#fff;background:rgba(0,0,0,.38);overflow:hidden;padding:0 16px;position:absolute;left:0;right:0}.mat-grid-tile .mat-grid-tile-footer>*,.mat-grid-tile .mat-grid-tile-header>*{margin:0;padding:0;font-weight:400;font-size:inherit}.mat-grid-tile .mat-grid-tile-footer.mat-2-line,.mat-grid-tile .mat-grid-tile-header.mat-2-line{height:68px}.mat-grid-tile .mat-grid-list-text{display:flex;flex-direction:column;width:100%;box-sizing:border-box;overflow:hidden}.mat-grid-tile .mat-grid-list-text>*{margin:0;padding:0;font-weight:400;font-size:inherit}.mat-grid-tile .mat-grid-list-text:empty{display:none}.mat-grid-tile .mat-grid-tile-header{top:0}.mat-grid-tile .mat-grid-tile-footer{bottom:0}.mat-grid-tile .mat-grid-avatar{padding-right:16px}[dir=rtl] .mat-grid-tile .mat-grid-avatar{padding-right:0;padding-left:16px}.mat-grid-tile .mat-grid-avatar:empty{display:none}'];
export const RenderType_MatGridTile:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MatGridTile,
    data:{}});
export function View_MatGridTile_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'figure',[['class',
      'mat-figure']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵncd((null as any),0)],(null as any),(null as any));
}
export function View_MatGridTile_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'mat-grid-tile',
      [['class','mat-grid-tile']],(null as any),(null as any),(null as any),View_MatGridTile_0,
      RenderType_MatGridTile)),i0.ɵdid(1,49152,(null as any),0,i1.MatGridTile,[i0.Renderer2,
      i0.ElementRef],(null as any),(null as any))],(null as any),(null as any));
}
export const MatGridTileNgFactory:i0.ComponentFactory<i1.MatGridTile> = i0.ɵccf('mat-grid-tile',
    i1.MatGridTile,View_MatGridTile_Host_0,{rowspan:'rowspan',colspan:'colspan'},{},
    ['*']);
const styles_ɵa13:any[] = ([] as any[]);
export const RenderType_ɵa13:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_ɵa13,
    data:{}});
export function View_ɵa13_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵncd((null as any),0),(_l()(),i0.ɵeld(1,0,(null as any),(null as any),
      1,'div',[['class','mat-grid-list-text']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),i0.ɵncd((null as any),1),i0.ɵncd((null as any),
      2)],(null as any),(null as any));
}
export function View_ɵa13_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),2,'mat-grid-tile-header',
      ([] as any[]),(null as any),(null as any),(null as any),View_ɵa13_0,RenderType_ɵa13)),
      i0.ɵdid(1,1097728,(null as any),1,i1.ɵa13,[i0.Renderer2,i0.ElementRef],(null as any),
          (null as any)),i0.ɵqud(603979776,1,{_lines:1})],(null as any),(null as any));
}
export const ɵa13NgFactory:i0.ComponentFactory<i1.ɵa13> = i0.ɵccf('mat-grid-tile-header, mat-grid-tile-footer',
    i1.ɵa13,View_ɵa13_Host_0,{},{},['[mat-grid-avatar], [matGridAvatar]','[mat-line], [matLine]',
        '*']);
