// @angular/material/card/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/material/card';
import * as i2 from '@angular/cdk/bidi';
import * as i3 from '@angular/platform-browser';
import * as i4 from '@angular/material/core';
export const MatCardModuleNgFactory:i0.NgModuleFactory<i1.MatCardModule> = i0.ɵcmf(i1.MatCardModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(6144,
          i2.DIR_DOCUMENT,(null as any),[i3.DOCUMENT]),i0.ɵmpd(4608,i2.Directionality,
          i2.Directionality,[[2,i2.DIR_DOCUMENT]]),i0.ɵmpd(512,i4.CompatibilityModule,
          i4.CompatibilityModule,([] as any[])),i0.ɵmpd(512,i2.BidiModule,i2.BidiModule,
          ([] as any[])),i0.ɵmpd(256,i4.MATERIAL_SANITY_CHECKS,true,([] as any[])),
          i0.ɵmpd(512,i4.MatCommonModule,i4.MatCommonModule,[[2,i4.MATERIAL_SANITY_CHECKS]]),
          i0.ɵmpd(512,i1.MatCardModule,i1.MatCardModule,([] as any[]))]);
    });
const styles_MatCard:any[] = ['.mat-card{transition:box-shadow 280ms cubic-bezier(.4,0,.2,1);display:block;position:relative;padding:24px;border-radius:2px}.mat-card:not([class*=mat-elevation-z]){box-shadow:0 3px 1px -2px rgba(0,0,0,.2),0 2px 2px 0 rgba(0,0,0,.14),0 1px 5px 0 rgba(0,0,0,.12)}@media screen and (-ms-high-contrast:active){.mat-card{outline:solid 1px}}.mat-card-flat{box-shadow:none}.mat-card-actions,.mat-card-content,.mat-card-subtitle,.mat-card-title{display:block;margin-bottom:16px}.mat-card-actions{margin-left:-16px;margin-right:-16px;padding:8px 0}.mat-card-actions-align-end{display:flex;justify-content:flex-end}.mat-card-image{width:calc(100% + 48px);margin:0 -24px 16px -24px}.mat-card-xl-image{width:240px;height:240px;margin:-8px}.mat-card-footer{display:block;margin:0 -24px -24px -24px}.mat-card-actions .mat-button,.mat-card-actions .mat-raised-button{margin:0 4px}.mat-card-header{display:flex;flex-direction:row}.mat-card-header-text{margin:0 8px}.mat-card-avatar{height:40px;width:40px;border-radius:50%;flex-shrink:0}.mat-card-lg-image,.mat-card-md-image,.mat-card-sm-image{margin:-8px 0}.mat-card-title-group{display:flex;justify-content:space-between;margin:0 -8px}.mat-card-sm-image{width:80px;height:80px}.mat-card-md-image{width:112px;height:112px}.mat-card-lg-image{width:152px;height:152px}@media (max-width:600px){.mat-card{padding:24px 16px}.mat-card-actions{margin-left:-8px;margin-right:-8px}.mat-card-image{width:calc(100% + 32px);margin:16px -16px}.mat-card-title-group{margin:0}.mat-card-xl-image{margin-left:0;margin-right:0}.mat-card-header{margin:-8px 0 0 0}.mat-card-footer{margin-left:-16px;margin-right:-16px}}.mat-card-content>:first-child,.mat-card>:first-child{margin-top:0}.mat-card-content>:last-child:not(.mat-card-footer),.mat-card>:last-child:not(.mat-card-footer){margin-bottom:0}.mat-card-image:first-child{margin-top:-24px}.mat-card>.mat-card-actions:last-child{margin-bottom:-16px;padding-bottom:0}.mat-card-actions .mat-button:first-child,.mat-card-actions .mat-raised-button:first-child{margin-left:0;margin-right:0}.mat-card-subtitle:not(:first-child),.mat-card-title:not(:first-child){margin-top:-4px}.mat-card-header .mat-card-subtitle:not(:first-child){margin-top:-8px}.mat-card>.mat-card-xl-image:first-child{margin-top:-8px}.mat-card>.mat-card-xl-image:last-child{margin-bottom:-8px}'];
export const RenderType_MatCard:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MatCard,
    data:{}});
export function View_MatCard_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵncd((null as any),0),i0.ɵncd((null as any),1)],(null as any),
      (null as any));
}
export function View_MatCard_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'mat-card',[['class',
      'mat-card']],(null as any),(null as any),(null as any),View_MatCard_0,RenderType_MatCard)),
      i0.ɵdid(1,49152,(null as any),0,i1.MatCard,([] as any[]),(null as any),(null as any))],
      (null as any),(null as any));
}
export const MatCardNgFactory:i0.ComponentFactory<i1.MatCard> = i0.ɵccf('mat-card',
    i1.MatCard,View_MatCard_Host_0,{},{},['*','mat-card-footer']);
const styles_MatCardHeader:any[] = ([] as any[]);
export const RenderType_MatCardHeader:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_MatCardHeader,data:{}});
export function View_MatCardHeader_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵncd((null as any),0),(_l()(),i0.ɵeld(1,0,(null as any),(null as any),
      1,'div',[['class','mat-card-header-text']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),i0.ɵncd((null as any),1),i0.ɵncd((null as any),
      2)],(null as any),(null as any));
}
export function View_MatCardHeader_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'mat-card-header',
      [['class','mat-card-header']],(null as any),(null as any),(null as any),View_MatCardHeader_0,
      RenderType_MatCardHeader)),i0.ɵdid(1,49152,(null as any),0,i1.MatCardHeader,
      ([] as any[]),(null as any),(null as any))],(null as any),(null as any));
}
export const MatCardHeaderNgFactory:i0.ComponentFactory<i1.MatCardHeader> = i0.ɵccf('mat-card-header',
    i1.MatCardHeader,View_MatCardHeader_Host_0,{},{},['[mat-card-avatar], [matCardAvatar]',
        'mat-card-title, mat-card-subtitle, [mat-card-title], [mat-card-subtitle]',
        '*']);
const styles_MatCardTitleGroup:any[] = ([] as any[]);
export const RenderType_MatCardTitleGroup:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_MatCardTitleGroup,data:{}});
export function View_MatCardTitleGroup_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'div',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),i0.ɵncd((null as any),
      0),i0.ɵncd((null as any),1),i0.ɵncd((null as any),2)],(null as any),(null as any));
}
export function View_MatCardTitleGroup_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'mat-card-title-group',
      [['class','mat-card-title-group']],(null as any),(null as any),(null as any),
      View_MatCardTitleGroup_0,RenderType_MatCardTitleGroup)),i0.ɵdid(1,49152,(null as any),
      0,i1.MatCardTitleGroup,([] as any[]),(null as any),(null as any))],(null as any),
      (null as any));
}
export const MatCardTitleGroupNgFactory:i0.ComponentFactory<i1.MatCardTitleGroup> = i0.ɵccf('mat-card-title-group',
    i1.MatCardTitleGroup,View_MatCardTitleGroup_Host_0,{},{},['mat-card-title, mat-card-subtitle, [mat-card-title], [mat-card-subtitle]',
        'img','*']);
