// @angular/material/core/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/material/core';
import * as i2 from '@angular/cdk/bidi';
import * as i3 from '@angular/platform-browser';
import * as i4 from '@angular/cdk/platform';
import * as i5 from '@angular/common';
export const MatCommonModuleNgFactory:i0.NgModuleFactory<i1.MatCommonModule> = i0.ɵcmf(i1.MatCommonModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(6144,
          i2.DIR_DOCUMENT,(null as any),[i3.DOCUMENT]),i0.ɵmpd(4608,i2.Directionality,
          i2.Directionality,[[2,i2.DIR_DOCUMENT]]),i0.ɵmpd(512,i1.CompatibilityModule,
          i1.CompatibilityModule,([] as any[])),i0.ɵmpd(512,i2.BidiModule,i2.BidiModule,
          ([] as any[])),i0.ɵmpd(256,i1.MATERIAL_SANITY_CHECKS,true,([] as any[])),
          i0.ɵmpd(512,i1.MatCommonModule,i1.MatCommonModule,[[2,i1.MATERIAL_SANITY_CHECKS]])]);
    });
export const CompatibilityModuleNgFactory:i0.NgModuleFactory<i1.CompatibilityModule> = i0.ɵcmf(i1.CompatibilityModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(512,
          i1.CompatibilityModule,i1.CompatibilityModule,([] as any[]))]);
    });
export const NoConflictStyleCompatibilityModeNgFactory:i0.NgModuleFactory<i1.NoConflictStyleCompatibilityMode> = i0.ɵcmf(i1.NoConflictStyleCompatibilityMode,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(512,
          i1.NoConflictStyleCompatibilityMode,i1.NoConflictStyleCompatibilityMode,
          ([] as any[]))]);
    });
export const NativeDateModuleNgFactory:i0.NgModuleFactory<i1.NativeDateModule> = i0.ɵcmf(i1.NativeDateModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(6144,
          i1.MAT_DATE_LOCALE,(null as any),[i0.LOCALE_ID]),i0.ɵmpd(4608,i1.DateAdapter,
          i1.NativeDateAdapter,[[2,i1.MAT_DATE_LOCALE]]),i0.ɵmpd(512,i1.NativeDateModule,
          i1.NativeDateModule,([] as any[]))]);
    });
export const MatNativeDateModuleNgFactory:i0.NgModuleFactory<i1.MatNativeDateModule> = i0.ɵcmf(i1.MatNativeDateModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(6144,
          i1.MAT_DATE_LOCALE,(null as any),[i0.LOCALE_ID]),i0.ɵmpd(4608,i1.DateAdapter,
          i1.NativeDateAdapter,[[2,i1.MAT_DATE_LOCALE]]),i0.ɵmpd(512,i1.NativeDateModule,
          i1.NativeDateModule,([] as any[])),i0.ɵmpd(512,i1.MatNativeDateModule,i1.MatNativeDateModule,
          ([] as any[])),i0.ɵmpd(256,i1.MAT_DATE_FORMATS,i1.MAT_NATIVE_DATE_FORMATS,
          ([] as any[]))]);
    });
export const MatLineModuleNgFactory:i0.NgModuleFactory<i1.MatLineModule> = i0.ɵcmf(i1.MatLineModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(6144,
          i2.DIR_DOCUMENT,(null as any),[i3.DOCUMENT]),i0.ɵmpd(4608,i2.Directionality,
          i2.Directionality,[[2,i2.DIR_DOCUMENT]]),i0.ɵmpd(512,i1.CompatibilityModule,
          i1.CompatibilityModule,([] as any[])),i0.ɵmpd(512,i2.BidiModule,i2.BidiModule,
          ([] as any[])),i0.ɵmpd(256,i1.MATERIAL_SANITY_CHECKS,true,([] as any[])),
          i0.ɵmpd(512,i1.MatCommonModule,i1.MatCommonModule,[[2,i1.MATERIAL_SANITY_CHECKS]]),
          i0.ɵmpd(512,i1.MatLineModule,i1.MatLineModule,([] as any[]))]);
    });
export const MatOptionModuleNgFactory:i0.NgModuleFactory<i1.MatOptionModule> = i0.ɵcmf(i1.MatOptionModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(6144,
          i2.DIR_DOCUMENT,(null as any),[i3.DOCUMENT]),i0.ɵmpd(4608,i2.Directionality,
          i2.Directionality,[[2,i2.DIR_DOCUMENT]]),i0.ɵmpd(4608,i4.Platform,i4.Platform,
          ([] as any[])),i0.ɵmpd(4608,i5.NgLocalization,i5.NgLocaleLocalization,[i0.LOCALE_ID]),
          i0.ɵmpd(512,i1.CompatibilityModule,i1.CompatibilityModule,([] as any[])),
          i0.ɵmpd(512,i2.BidiModule,i2.BidiModule,([] as any[])),i0.ɵmpd(256,i1.MATERIAL_SANITY_CHECKS,
              true,([] as any[])),i0.ɵmpd(512,i1.MatCommonModule,i1.MatCommonModule,
              [[2,i1.MATERIAL_SANITY_CHECKS]]),i0.ɵmpd(512,i4.PlatformModule,i4.PlatformModule,
              ([] as any[])),i0.ɵmpd(512,i1.MatRippleModule,i1.MatRippleModule,([] as any[])),
          i0.ɵmpd(512,i5.CommonModule,i5.CommonModule,([] as any[])),i0.ɵmpd(512,i1.MatPseudoCheckboxModule,
              i1.MatPseudoCheckboxModule,([] as any[])),i0.ɵmpd(512,i1.MatOptionModule,
              i1.MatOptionModule,([] as any[]))]);
    });
export const MatRippleModuleNgFactory:i0.NgModuleFactory<i1.MatRippleModule> = i0.ɵcmf(i1.MatRippleModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(6144,
          i2.DIR_DOCUMENT,(null as any),[i3.DOCUMENT]),i0.ɵmpd(4608,i2.Directionality,
          i2.Directionality,[[2,i2.DIR_DOCUMENT]]),i0.ɵmpd(4608,i4.Platform,i4.Platform,
          ([] as any[])),i0.ɵmpd(512,i1.CompatibilityModule,i1.CompatibilityModule,
          ([] as any[])),i0.ɵmpd(512,i2.BidiModule,i2.BidiModule,([] as any[])),i0.ɵmpd(256,
          i1.MATERIAL_SANITY_CHECKS,true,([] as any[])),i0.ɵmpd(512,i1.MatCommonModule,
          i1.MatCommonModule,[[2,i1.MATERIAL_SANITY_CHECKS]]),i0.ɵmpd(512,i4.PlatformModule,
          i4.PlatformModule,([] as any[])),i0.ɵmpd(512,i1.MatRippleModule,i1.MatRippleModule,
          ([] as any[]))]);
    });
export const MatPseudoCheckboxModuleNgFactory:i0.NgModuleFactory<i1.MatPseudoCheckboxModule> = i0.ɵcmf(i1.MatPseudoCheckboxModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(512,
          i1.MatPseudoCheckboxModule,i1.MatPseudoCheckboxModule,([] as any[]))]);
    });
const styles_MatOption:any[] = ([] as any[]);
export const RenderType_MatOption:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MatOption,
    data:{}});
function View_MatOption_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),3,'span',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵeld(1,0,(null as any),(null as any),2,'mat-pseudo-checkbox',[['class','mat-option-pseudo-checkbox mat-pseudo-checkbox']],
          [[2,'mat-pseudo-checkbox-indeterminate',(null as any)],[2,'mat-pseudo-checkbox-checked',
              (null as any)],[2,'mat-pseudo-checkbox-disabled',(null as any)]],(null as any),
          (null as any),View_MatPseudoCheckbox_0,RenderType_MatPseudoCheckbox)),i0.ɵdid(2,
      16384,(null as any),0,i1.MatPrefixRejector,([] as any[]),(null as any),(null as any)),
      i0.ɵdid(3,49152,(null as any),0,i1.MatPseudoCheckbox,([] as any[]),{state:[0,
          'state'],disabled:[1,'disabled']},(null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_3:any = (_co.selected? 'checked': '');
    const currVal_4:any = _co.disabled;
    _ck(_v,3,0,currVal_3,currVal_4);
  },(_ck,_v) => {
    const currVal_0:any = (i0.ɵnov(_v,3).state === 'indeterminate');
    const currVal_1:any = (i0.ɵnov(_v,3).state === 'checked');
    const currVal_2:any = i0.ɵnov(_v,3).disabled;
    _ck(_v,1,0,currVal_0,currVal_1,currVal_2);
  });
}
export function View_MatOption_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_MatOption_1)),i0.ɵdid(1,16384,(null as any),0,i5.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵeld(2,0,(null as any),
      (null as any),1,'span',[['class','mat-option-text']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),i0.ɵncd((null as any),0),(_l()(),
      i0.ɵeld(4,0,(null as any),(null as any),1,'div',[['class','mat-option-ripple mat-ripple'],
          ['mat-ripple','']],[[2,'mat-ripple-unbounded',(null as any)]],(null as any),
          (null as any),(null as any),(null as any))),i0.ɵdid(5,671744,(null as any),
      0,i1.MatRipple,[i0.ElementRef,i0.NgZone,i4.Platform,[2,i1.MAT_RIPPLE_GLOBAL_OPTIONS]],
      {trigger:[0,'trigger'],disabled:[1,'disabled']},(null as any))],(_ck,_v) => {
    var _co:i1.MatOption = _v.component;
    const currVal_0:any = _co.multiple;
    _ck(_v,1,0,currVal_0);
    const currVal_2:any = _co._getHostElement();
    const currVal_3:any = (_co.disabled || _co.disableRipple);
    _ck(_v,5,0,currVal_2,currVal_3);
  },(_ck,_v) => {
    const currVal_1:any = i0.ɵnov(_v,5).unbounded;
    _ck(_v,4,0,currVal_1);
  });
}
export function View_MatOption_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'mat-option',
      [['class','mat-option'],['role','option']],[[1,'tabindex',0],[2,'mat-selected',
          (null as any)],[2,'mat-option-multiple',(null as any)],[2,'mat-active',(null as any)],
          [8,'id',0],[1,'aria-selected',0],[1,'aria-disabled',0],[2,'mat-option-disabled',
              (null as any)]],[[(null as any),'click'],[(null as any),'keydown']],
      (_v,en,$event) => {
        var ad:boolean = true;
        if (('click' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,1)._selectViaInteraction()) !== false);
          ad = (pd_0 && ad);
        }
        if (('keydown' === en)) {
          const pd_1:any = ((<any>i0.ɵnov(_v,1)._handleKeydown($event)) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },View_MatOption_0,RenderType_MatOption)),i0.ɵdid(1,49152,(null as any),0,i1.MatOption,
      [i0.ElementRef,i0.ChangeDetectorRef,[2,i1.MatOptgroup]],(null as any),(null as any))],
      (null as any),(_ck,_v) => {
        const currVal_0:any = i0.ɵnov(_v,1)._getTabIndex();
        const currVal_1:any = i0.ɵnov(_v,1).selected;
        const currVal_2:any = i0.ɵnov(_v,1).multiple;
        const currVal_3:any = i0.ɵnov(_v,1).active;
        const currVal_4:any = i0.ɵnov(_v,1).id;
        const currVal_5:any = i0.ɵnov(_v,1).selected.toString();
        const currVal_6:any = i0.ɵnov(_v,1).disabled.toString();
        const currVal_7:any = i0.ɵnov(_v,1).disabled;
        _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,
            currVal_7);
      });
}
export const MatOptionNgFactory:i0.ComponentFactory<i1.MatOption> = i0.ɵccf('mat-option',
    i1.MatOption,View_MatOption_Host_0,{value:'value',disabled:'disabled'},{onSelectionChange:'onSelectionChange'},
    ['*']);
const styles_MatOptgroup:any[] = ([] as any[]);
export const RenderType_MatOptgroup:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MatOptgroup,
    data:{}});
export function View_MatOptgroup_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'label',[['class',
      'mat-optgroup-label']],[[8,'id',0]],(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted(1,(null as any),['',''])),i0.ɵncd((null as any),
      0)],(null as any),(_ck,_v) => {
    var _co:i1.MatOptgroup = _v.component;
    const currVal_0:any = _co._labelId;
    _ck(_v,0,0,currVal_0);
    const currVal_1:any = _co.label;
    _ck(_v,1,0,currVal_1);
  });
}
export function View_MatOptgroup_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'mat-optgroup',
      [['class','mat-optgroup'],['role','group']],[[2,'mat-optgroup-disabled',(null as any)],
          [1,'aria-disabled',0],[1,'aria-labelledby',0]],(null as any),(null as any),
      View_MatOptgroup_0,RenderType_MatOptgroup)),i0.ɵdid(1,49152,(null as any),0,
      i1.MatOptgroup,([] as any[]),(null as any),(null as any))],(null as any),(_ck,
      _v) => {
    const currVal_0:any = i0.ɵnov(_v,1).disabled;
    const currVal_1:any = i0.ɵnov(_v,1).disabled.toString();
    const currVal_2:any = i0.ɵnov(_v,1)._labelId;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export const MatOptgroupNgFactory:i0.ComponentFactory<i1.MatOptgroup> = i0.ɵccf('mat-optgroup',
    i1.MatOptgroup,View_MatOptgroup_Host_0,{disabled:'disabled',label:'label'},{},
    ['mat-option']);
const styles_MatPseudoCheckbox:any[] = ['.mat-pseudo-checkbox{width:20px;height:20px;border:2px solid;border-radius:2px;cursor:pointer;display:inline-block;vertical-align:middle;box-sizing:border-box;position:relative;flex-shrink:0;transition:border-color 90ms cubic-bezier(0,0,.2,.1),background-color 90ms cubic-bezier(0,0,.2,.1)}.mat-pseudo-checkbox::after{position:absolute;opacity:0;content:\'\';border-bottom:2px solid currentColor;transition:opacity 90ms cubic-bezier(0,0,.2,.1)}.mat-pseudo-checkbox.mat-pseudo-checkbox-checked,.mat-pseudo-checkbox.mat-pseudo-checkbox-indeterminate{border:none}.mat-pseudo-checkbox-disabled{cursor:default}.mat-pseudo-checkbox-indeterminate::after{top:9px;left:2px;width:16px;opacity:1}.mat-pseudo-checkbox-checked::after{top:5px;left:3px;width:12px;height:5px;border-left:2px solid currentColor;transform:rotate(-45deg);opacity:1}'];
export const RenderType_MatPseudoCheckbox:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_MatPseudoCheckbox,data:{}});
export function View_MatPseudoCheckbox_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,([] as any[]),(null as any),(null as any));
}
export function View_MatPseudoCheckbox_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),1,'mat-pseudo-checkbox',
      [['class','mat-pseudo-checkbox']],[[2,'mat-pseudo-checkbox-indeterminate',(null as any)],
          [2,'mat-pseudo-checkbox-checked',(null as any)],[2,'mat-pseudo-checkbox-disabled',
              (null as any)]],(null as any),(null as any),View_MatPseudoCheckbox_0,
      RenderType_MatPseudoCheckbox)),i0.ɵdid(1,49152,(null as any),0,i1.MatPseudoCheckbox,
      ([] as any[]),(null as any),(null as any))],(null as any),(_ck,_v) => {
    const currVal_0:any = (i0.ɵnov(_v,1).state === 'indeterminate');
    const currVal_1:any = (i0.ɵnov(_v,1).state === 'checked');
    const currVal_2:any = i0.ɵnov(_v,1).disabled;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export const MatPseudoCheckboxNgFactory:i0.ComponentFactory<i1.MatPseudoCheckbox> = i0.ɵccf('mat-pseudo-checkbox',
    i1.MatPseudoCheckbox,View_MatPseudoCheckbox_Host_0,{state:'state',disabled:'disabled'},
    {},([] as any[]));
