// @angular/material/sidenav/typings/index.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from '@angular/material/sidenav';
import * as i2 from '@angular/common';
import * as i3 from '@angular/cdk/bidi';
import * as i4 from '@angular/platform-browser';
import * as i5 from '@angular/cdk/platform';
import * as i6 from '@angular/cdk/a11y';
import * as i7 from '@angular/cdk/scrolling';
import * as i8 from '@angular/cdk/overlay';
import * as i9 from '@angular/material/core';
import * as i10 from '@angular/cdk/portal';
export var MatSidenavModuleNgFactory = i0.ɵcmf(i1.MatSidenavModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(6144, i3.DIR_DOCUMENT, null, [i4.DOCUMENT]), i0.ɵmpd(4608, i3.Directionality, i3.Directionality, [[2, i3.DIR_DOCUMENT]]), i0.ɵmpd(4608, i5.Platform, i5.Platform, []),
        i0.ɵmpd(4608, i6.InteractivityChecker, i6.InteractivityChecker, [i5.Platform]),
        i0.ɵmpd(4608, i6.FocusTrapFactory, i6.FocusTrapFactory, [i6.InteractivityChecker,
            i5.Platform, i0.NgZone]), i0.ɵmpd(136192, i6.AriaDescriber, i6.ARIA_DESCRIBER_PROVIDER_FACTORY, [[3, i6.AriaDescriber], i5.Platform]), i0.ɵmpd(5120, i6.LiveAnnouncer, i6.LIVE_ANNOUNCER_PROVIDER_FACTORY, [[3, i6.LiveAnnouncer], [2, i6.LIVE_ANNOUNCER_ELEMENT_TOKEN], i5.Platform]),
        i0.ɵmpd(5120, i6.FocusMonitor, i6.FOCUS_MONITOR_PROVIDER_FACTORY, [[3, i6.FocusMonitor],
            i0.NgZone, i5.Platform]), i0.ɵmpd(5120, i7.ScrollDispatcher, i7.SCROLL_DISPATCHER_PROVIDER_FACTORY, [[3, i7.ScrollDispatcher], i0.NgZone, i5.Platform]), i0.ɵmpd(5120, i7.ViewportRuler, i7.VIEWPORT_RULER_PROVIDER_FACTORY, [[3, i7.ViewportRuler], i5.Platform,
            i0.NgZone, i7.ScrollDispatcher]), i0.ɵmpd(4608, i8.ScrollStrategyOptions, i8.ScrollStrategyOptions, [i7.ScrollDispatcher, i7.ViewportRuler]), i0.ɵmpd(5120, i8.OverlayContainer, i8.ɵa, [[3, i8.OverlayContainer]]), i0.ɵmpd(4608, i8.ɵf, i8.ɵf, [i7.ViewportRuler]), i0.ɵmpd(4608, i8.Overlay, i8.Overlay, [i8.ScrollStrategyOptions,
            i8.OverlayContainer, i0.ComponentFactoryResolver, i8.ɵf, i0.ApplicationRef,
            i0.Injector, i0.NgZone]), i0.ɵmpd(5120, i8.ɵc, i8.ɵd, [i8.Overlay]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i9.CompatibilityModule, i9.CompatibilityModule, []), i0.ɵmpd(512, i3.BidiModule, i3.BidiModule, []), i0.ɵmpd(256, i9.MATERIAL_SANITY_CHECKS, true, []),
        i0.ɵmpd(512, i9.MatCommonModule, i9.MatCommonModule, [[2, i9.MATERIAL_SANITY_CHECKS]]),
        i0.ɵmpd(512, i5.PlatformModule, i5.PlatformModule, []), i0.ɵmpd(512, i6.A11yModule, i6.A11yModule, []), i0.ɵmpd(512, i10.PortalModule, i10.PortalModule, []), i0.ɵmpd(512, i7.ScrollDispatchModule, i7.ScrollDispatchModule, []), i0.ɵmpd(512, i8.OverlayModule, i8.OverlayModule, []), i0.ɵmpd(512, i1.MatSidenavModule, i1.MatSidenavModule, [])]);
});
var styles_MatDrawer = [];
export var RenderType_MatDrawer = i0.ɵcrt({ encapsulation: 2, styles: styles_MatDrawer,
    data: { 'animation': [{ type: 7, name: 'transform', definitions: [{ type: 0, name: 'open, open-instant',
                        styles: { type: 6, styles: { transform: 'translate3d(0, 0, 0)', visibility: 'visible' },
                            offset: null }, options: undefined }, { type: 0, name: 'void',
                        styles: { type: 6, styles: { visibility: 'hidden' }, offset: null }, options: undefined },
                    { type: 1, expr: 'void => open-instant', animation: { type: 4, styles: null,
                            timings: '0ms' }, options: null }, { type: 1, expr: 'void <=> open, open-instant => void',
                        animation: { type: 4, styles: null, timings: '400ms cubic-bezier(0.25, 0.8, 0.25, 1)' },
                        options: null }], options: {} }] } });
export function View_MatDrawer_0(_l) {
    return i0.ɵvid(2, [i0.ɵncd(null, 0)], null, null);
}
export function View_MatDrawer_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'mat-drawer', [['class', 'mat-drawer'], ['tabIndex', '-1']], [[40, '@transform', 0], [1, 'align', 0],
            [2, 'mat-drawer-end', null], [2, 'mat-drawer-over', null], [2,
                'mat-drawer-push', null], [2, 'mat-drawer-side', null]], [['component', '@transform.start'], ['component', '@transform.done'], [null,
                'keydown']], function (_v, en, $event) {
            var ad = true;
            if (('component:@transform.start' === en)) {
                var pd_0 = (i0.ɵnov(_v, 1)._onAnimationStart($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('component:@transform.done' === en)) {
                var pd_1 = (i0.ɵnov(_v, 1)._onAnimationEnd($event) !== false);
                ad = (pd_1 && ad);
            }
            if (('keydown' === en)) {
                var pd_2 = (i0.ɵnov(_v, 1).handleKeydown($event) !== false);
                ad = (pd_2 && ad);
            }
            return ad;
        }, View_MatDrawer_0, RenderType_MatDrawer)), i0.ɵdid(1, 1228800, null, 0, i1.MatDrawer, [i0.ElementRef, i6.FocusTrapFactory, [2, i4.DOCUMENT]], null, null)], null, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 1)._animationState;
        var currVal_1 = null;
        var currVal_2 = (i0.ɵnov(_v, 1).position === 'end');
        var currVal_3 = (i0.ɵnov(_v, 1).mode === 'over');
        var currVal_4 = (i0.ɵnov(_v, 1).mode === 'push');
        var currVal_5 = (i0.ɵnov(_v, 1).mode === 'side');
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5);
    });
}
export var MatDrawerNgFactory = i0.ɵccf('mat-drawer', i1.MatDrawer, View_MatDrawer_Host_0, { position: 'position', align: 'align', mode: 'mode',
    disableClose: 'disableClose', opened: 'opened' }, { onOpen: 'open', onClose: 'close',
    onPositionChanged: 'positionChanged', onAlignChanged: 'align-changed' }, ['*']);
var styles_MatDrawerContainer = ['.mat-drawer-container{position:relative;z-index:1;box-sizing:border-box;-webkit-overflow-scrolling:touch;display:block;overflow:hidden}.mat-drawer-container[fullscreen]{top:0;left:0;right:0;bottom:0;position:absolute}.mat-drawer-container[fullscreen].mat-drawer-opened{overflow:hidden}.mat-drawer-backdrop{top:0;left:0;right:0;bottom:0;position:absolute;display:block;z-index:3;visibility:hidden}.mat-drawer-backdrop.mat-drawer-shown{visibility:visible}.mat-drawer-transition .mat-drawer-backdrop{transition-duration:.4s;transition-timing-function:cubic-bezier(.25,.8,.25,1);transition-property:background-color,visibility}@media screen and (-ms-high-contrast:active){.mat-drawer-backdrop{opacity:.5}}.mat-drawer-content{position:relative;z-index:1;display:block;height:100%;overflow:auto}.mat-drawer-transition .mat-drawer-content{transition-duration:.4s;transition-timing-function:cubic-bezier(.25,.8,.25,1);transition-property:transform,margin-left,margin-right}.mat-drawer{position:relative;z-index:4;display:block;position:absolute;top:0;bottom:0;z-index:3;min-width:5vw;outline:0;box-sizing:border-box;overflow-y:auto;transform:translate3d(-100%,0,0)}.mat-drawer.mat-drawer-side{z-index:2}.mat-drawer.mat-drawer-end{right:0;transform:translate3d(100%,0,0)}[dir=rtl] .mat-drawer{transform:translate3d(100%,0,0)}[dir=rtl] .mat-drawer.mat-drawer-end{left:0;right:auto;transform:translate3d(-100%,0,0)}.mat-drawer.mat-drawer-opened:not(.mat-drawer-side),.mat-drawer.mat-drawer-opening:not(.mat-drawer-side){box-shadow:0 8px 10px -5px rgba(0,0,0,.2),0 16px 24px 2px rgba(0,0,0,.14),0 6px 30px 5px rgba(0,0,0,.12)}.mat-sidenav-fixed{position:fixed}'];
export var RenderType_MatDrawerContainer = i0.ɵcrt({ encapsulation: 2,
    styles: styles_MatDrawerContainer, data: {} });
function View_MatDrawerContainer_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 3, 'mat-drawer-content', [['cdkScrollable', ''], ['class', 'mat-drawer-content']], [[4, 'marginLeft', 'px'],
            [4, 'marginRight', 'px']], null, null, View_MatDrawerContent_0, RenderType_MatDrawerContent)), i0.ɵdid(1, 212992, null, 0, i7.Scrollable, [i0.ElementRef, i7.ScrollDispatcher, i0.NgZone, i0.Renderer2], null, null),
        i0.ɵdid(2, 1097728, null, 0, i1.MatDrawerContent, [i0.ChangeDetectorRef,
            i1.MatDrawerContainer], null, null), i0.ɵncd(0, 2)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 2)._margins.left;
        var currVal_1 = i0.ɵnov(_v, 2)._margins.right;
        _ck(_v, 0, 0, currVal_0, currVal_1);
    });
}
export function View_MatDrawerContainer_0(_l) {
    return i0.ɵvid(2, [(_l()(), i0.ɵeld(0, 0, null, null, 0, 'div', [['class',
                'mat-drawer-backdrop']], [[2, 'mat-drawer-shown', null]], [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co._onBackdropClicked() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵncd(null, 0), i0.ɵncd(null, 1),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_MatDrawerContainer_1)),
        i0.ɵdid(4, 16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = !_co._content;
        _ck(_v, 4, 0, currVal_1);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co._isShowingBackdrop();
        _ck(_v, 0, 0, currVal_0);
    });
}
export function View_MatDrawerContainer_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 3, 'mat-drawer-container', [['class', 'mat-drawer-container']], null, null, null, View_MatDrawerContainer_0, RenderType_MatDrawerContainer)), i0.ɵdid(1, 1228800, null, 2, i1.MatDrawerContainer, [[2, i3.Directionality], i0.ElementRef, i0.Renderer2,
            i0.NgZone, i0.ChangeDetectorRef], null, null), i0.ɵqud(603979776, 1, { _drawers: 1 }), i0.ɵqud(335544320, 2, { _content: 0 })], null, null);
}
export var MatDrawerContainerNgFactory = i0.ɵccf('mat-drawer-container', i1.MatDrawerContainer, View_MatDrawerContainer_Host_0, {}, { backdropClick: 'backdropClick' }, ['mat-drawer', 'mat-drawer-content', '*']);
var styles_MatDrawerContent = [];
export var RenderType_MatDrawerContent = i0.ɵcrt({ encapsulation: 2,
    styles: styles_MatDrawerContent, data: {} });
export function View_MatDrawerContent_0(_l) {
    return i0.ɵvid(2, [i0.ɵncd(null, 0)], null, null);
}
export function View_MatDrawerContent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'mat-drawer-content', [['class', 'mat-drawer-content']], [[4, 'marginLeft', 'px'], [4, 'marginRight', 'px']], null, null, View_MatDrawerContent_0, RenderType_MatDrawerContent)),
        i0.ɵdid(1, 1097728, null, 0, i1.MatDrawerContent, [i0.ChangeDetectorRef,
            i1.MatDrawerContainer], null, null)], null, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 1)._margins.left;
        var currVal_1 = i0.ɵnov(_v, 1)._margins.right;
        _ck(_v, 0, 0, currVal_0, currVal_1);
    });
}
export var MatDrawerContentNgFactory = i0.ɵccf('mat-drawer-content', i1.MatDrawerContent, View_MatDrawerContent_Host_0, {}, {}, ['*']);
var styles_MatSidenav = [];
export var RenderType_MatSidenav = i0.ɵcrt({ encapsulation: 2, styles: styles_MatSidenav,
    data: { 'animation': [{ type: 7, name: 'transform', definitions: [{ type: 0, name: 'open, open-instant',
                        styles: { type: 6, styles: { transform: 'translate3d(0, 0, 0)', visibility: 'visible' },
                            offset: null }, options: undefined }, { type: 0, name: 'void',
                        styles: { type: 6, styles: { visibility: 'hidden' }, offset: null }, options: undefined },
                    { type: 1, expr: 'void => open-instant', animation: { type: 4, styles: null,
                            timings: '0ms' }, options: null }, { type: 1, expr: 'void <=> open, open-instant => void',
                        animation: { type: 4, styles: null, timings: '400ms cubic-bezier(0.25, 0.8, 0.25, 1)' },
                        options: null }], options: {} }] } });
export function View_MatSidenav_0(_l) {
    return i0.ɵvid(2, [i0.ɵncd(null, 0)], null, null);
}
export function View_MatSidenav_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'mat-sidenav', [['class', 'mat-drawer mat-sidenav'], ['tabIndex', '-1']], [[40, '@transform', 0],
            [1, 'align', 0], [2, 'mat-drawer-end', null], [2, 'mat-drawer-over', null],
            [2, 'mat-drawer-push', null], [2, 'mat-drawer-side', null],
            [2, 'mat-sidenav-fixed', null], [4, 'top', 'px'], [4, 'bottom', 'px']], [['component', '@transform.start'], ['component', '@transform.done'], [null,
                'keydown']], function (_v, en, $event) {
            var ad = true;
            if (('component:@transform.start' === en)) {
                var pd_0 = (i0.ɵnov(_v, 1)._onAnimationStart($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('component:@transform.done' === en)) {
                var pd_1 = (i0.ɵnov(_v, 1)._onAnimationEnd($event) !== false);
                ad = (pd_1 && ad);
            }
            if (('keydown' === en)) {
                var pd_2 = (i0.ɵnov(_v, 1).handleKeydown($event) !== false);
                ad = (pd_2 && ad);
            }
            return ad;
        }, View_MatSidenav_0, RenderType_MatSidenav)), i0.ɵdid(1, 1228800, null, 0, i1.MatSidenav, [i0.ElementRef, i6.FocusTrapFactory, [2, i4.DOCUMENT]], null, null)], null, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 1)._animationState;
        var currVal_1 = null;
        var currVal_2 = (i0.ɵnov(_v, 1).position === 'end');
        var currVal_3 = (i0.ɵnov(_v, 1).mode === 'over');
        var currVal_4 = (i0.ɵnov(_v, 1).mode === 'push');
        var currVal_5 = (i0.ɵnov(_v, 1).mode === 'side');
        var currVal_6 = i0.ɵnov(_v, 1).fixedInViewport;
        var currVal_7 = (i0.ɵnov(_v, 1).fixedInViewport ? i0.ɵnov(_v, 1).fixedTopGap : null);
        var currVal_8 = (i0.ɵnov(_v, 1).fixedInViewport ? i0.ɵnov(_v, 1).fixedBottomGap : null);
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5, currVal_6, currVal_7, currVal_8);
    });
}
export var MatSidenavNgFactory = i0.ɵccf('mat-sidenav', i1.MatSidenav, View_MatSidenav_Host_0, { position: 'position', align: 'align', mode: 'mode',
    disableClose: 'disableClose', opened: 'opened', fixedInViewport: 'fixedInViewport',
    fixedTopGap: 'fixedTopGap', fixedBottomGap: 'fixedBottomGap' }, { onOpen: 'open',
    onClose: 'close', onPositionChanged: 'positionChanged', onAlignChanged: 'align-changed' }, ['*']);
var styles_MatSidenavContainer = ['.mat-drawer-container{position:relative;z-index:1;box-sizing:border-box;-webkit-overflow-scrolling:touch;display:block;overflow:hidden}.mat-drawer-container[fullscreen]{top:0;left:0;right:0;bottom:0;position:absolute}.mat-drawer-container[fullscreen].mat-drawer-opened{overflow:hidden}.mat-drawer-backdrop{top:0;left:0;right:0;bottom:0;position:absolute;display:block;z-index:3;visibility:hidden}.mat-drawer-backdrop.mat-drawer-shown{visibility:visible}.mat-drawer-transition .mat-drawer-backdrop{transition-duration:.4s;transition-timing-function:cubic-bezier(.25,.8,.25,1);transition-property:background-color,visibility}@media screen and (-ms-high-contrast:active){.mat-drawer-backdrop{opacity:.5}}.mat-drawer-content{position:relative;z-index:1;display:block;height:100%;overflow:auto}.mat-drawer-transition .mat-drawer-content{transition-duration:.4s;transition-timing-function:cubic-bezier(.25,.8,.25,1);transition-property:transform,margin-left,margin-right}.mat-drawer{position:relative;z-index:4;display:block;position:absolute;top:0;bottom:0;z-index:3;min-width:5vw;outline:0;box-sizing:border-box;overflow-y:auto;transform:translate3d(-100%,0,0)}.mat-drawer.mat-drawer-side{z-index:2}.mat-drawer.mat-drawer-end{right:0;transform:translate3d(100%,0,0)}[dir=rtl] .mat-drawer{transform:translate3d(100%,0,0)}[dir=rtl] .mat-drawer.mat-drawer-end{left:0;right:auto;transform:translate3d(-100%,0,0)}.mat-drawer.mat-drawer-opened:not(.mat-drawer-side),.mat-drawer.mat-drawer-opening:not(.mat-drawer-side){box-shadow:0 8px 10px -5px rgba(0,0,0,.2),0 16px 24px 2px rgba(0,0,0,.14),0 6px 30px 5px rgba(0,0,0,.12)}.mat-sidenav-fixed{position:fixed}'];
export var RenderType_MatSidenavContainer = i0.ɵcrt({ encapsulation: 2,
    styles: styles_MatSidenavContainer, data: {} });
function View_MatSidenavContainer_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 3, 'mat-sidenav-content', [['cdkScrollable', ''], ['class', 'mat-drawer-content mat-sidenav-content']], [[4,
                'marginLeft', 'px'], [4, 'marginRight', 'px']], null, null, View_MatSidenavContent_0, RenderType_MatSidenavContent)), i0.ɵdid(1, 212992, null, 0, i7.Scrollable, [i0.ElementRef, i7.ScrollDispatcher, i0.NgZone, i0.Renderer2], null, null),
        i0.ɵdid(2, 1097728, null, 0, i1.MatSidenavContent, [i0.ChangeDetectorRef,
            i1.MatSidenavContainer], null, null), i0.ɵncd(0, 2)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 2)._margins.left;
        var currVal_1 = i0.ɵnov(_v, 2)._margins.right;
        _ck(_v, 0, 0, currVal_0, currVal_1);
    });
}
export function View_MatSidenavContainer_0(_l) {
    return i0.ɵvid(2, [(_l()(), i0.ɵeld(0, 0, null, null, 0, 'div', [['class',
                'mat-drawer-backdrop']], [[2, 'mat-drawer-shown', null]], [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co._onBackdropClicked() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵncd(null, 0), i0.ɵncd(null, 1),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_MatSidenavContainer_1)),
        i0.ɵdid(4, 16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = !_co._content;
        _ck(_v, 4, 0, currVal_1);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co._isShowingBackdrop();
        _ck(_v, 0, 0, currVal_0);
    });
}
export function View_MatSidenavContainer_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 3, 'mat-sidenav-container', [['class', 'mat-drawer-container mat-sidenav-container']], null, null, null, View_MatSidenavContainer_0, RenderType_MatSidenavContainer)), i0.ɵdid(1, 1228800, null, 2, i1.MatSidenavContainer, [[2, i3.Directionality], i0.ElementRef,
            i0.Renderer2, i0.NgZone, i0.ChangeDetectorRef], null, null),
        i0.ɵqud(603979776, 1, { _drawers: 1 }), i0.ɵqud(335544320, 2, { _content: 0 })], null, null);
}
export var MatSidenavContainerNgFactory = i0.ɵccf('mat-sidenav-container', i1.MatSidenavContainer, View_MatSidenavContainer_Host_0, {}, { backdropClick: 'backdropClick' }, ['mat-sidenav', 'mat-sidenav-content', '*']);
var styles_MatSidenavContent = [];
export var RenderType_MatSidenavContent = i0.ɵcrt({ encapsulation: 2,
    styles: styles_MatSidenavContent, data: {} });
export function View_MatSidenavContent_0(_l) {
    return i0.ɵvid(2, [i0.ɵncd(null, 0)], null, null);
}
export function View_MatSidenavContent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'mat-sidenav-content', [['class', 'mat-drawer-content mat-sidenav-content']], [[4, 'marginLeft', 'px'],
            [4, 'marginRight', 'px']], null, null, View_MatSidenavContent_0, RenderType_MatSidenavContent)), i0.ɵdid(1, 1097728, null, 0, i1.MatSidenavContent, [i0.ChangeDetectorRef, i1.MatSidenavContainer], null, null)], null, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 1)._margins.left;
        var currVal_1 = i0.ɵnov(_v, 1)._margins.right;
        _ck(_v, 0, 0, currVal_0, currVal_1);
    });
}
export var MatSidenavContentNgFactory = i0.ɵccf('mat-sidenav-content', i1.MatSidenavContent, View_MatSidenavContent_Host_0, {}, {}, ['*']);
