// @angular/material/autocomplete/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/material/autocomplete';
import * as i2 from '@angular/cdk/bidi';
import * as i3 from '@angular/platform-browser';
import * as i4 from '@angular/cdk/platform';
import * as i5 from '@angular/common';
import * as i6 from '@angular/cdk/scrolling';
import * as i7 from '@angular/cdk/overlay';
import * as i8 from '@angular/material/core';
import * as i9 from '@angular/cdk/portal';
export const MatAutocompleteModuleNgFactory:i0.NgModuleFactory<i1.MatAutocompleteModule> = i0.ɵcmf(i1.MatAutocompleteModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(6144,
          i2.DIR_DOCUMENT,(null as any),[i3.DOCUMENT]),i0.ɵmpd(4608,i2.Directionality,
          i2.Directionality,[[2,i2.DIR_DOCUMENT]]),i0.ɵmpd(4608,i4.Platform,i4.Platform,
          ([] as any[])),i0.ɵmpd(4608,i5.NgLocalization,i5.NgLocaleLocalization,[i0.LOCALE_ID]),
          i0.ɵmpd(5120,i6.ScrollDispatcher,i6.SCROLL_DISPATCHER_PROVIDER_FACTORY,[[3,
              i6.ScrollDispatcher],i0.NgZone,i4.Platform]),i0.ɵmpd(5120,i6.ViewportRuler,
              i6.VIEWPORT_RULER_PROVIDER_FACTORY,[[3,i6.ViewportRuler],i4.Platform,
                  i0.NgZone,i6.ScrollDispatcher]),i0.ɵmpd(4608,i7.ScrollStrategyOptions,
              i7.ScrollStrategyOptions,[i6.ScrollDispatcher,i6.ViewportRuler]),i0.ɵmpd(5120,
              i7.OverlayContainer,i7.ɵa,[[3,i7.OverlayContainer]]),i0.ɵmpd(4608,i7.ɵf,
              i7.ɵf,[i6.ViewportRuler]),i0.ɵmpd(4608,i7.Overlay,i7.Overlay,[i7.ScrollStrategyOptions,
              i7.OverlayContainer,i0.ComponentFactoryResolver,i7.ɵf,i0.ApplicationRef,
              i0.Injector,i0.NgZone]),i0.ɵmpd(5120,i7.ɵc,i7.ɵd,[i7.Overlay]),i0.ɵmpd(5120,
              i1.MAT_AUTOCOMPLETE_SCROLL_STRATEGY,i1.MAT_AUTOCOMPLETE_SCROLL_STRATEGY_PROVIDER_FACTORY,
              [i7.Overlay]),i0.ɵmpd(512,i8.CompatibilityModule,i8.CompatibilityModule,
              ([] as any[])),i0.ɵmpd(512,i2.BidiModule,i2.BidiModule,([] as any[])),
          i0.ɵmpd(256,i8.MATERIAL_SANITY_CHECKS,true,([] as any[])),i0.ɵmpd(512,i8.MatCommonModule,
              i8.MatCommonModule,[[2,i8.MATERIAL_SANITY_CHECKS]]),i0.ɵmpd(512,i4.PlatformModule,
              i4.PlatformModule,([] as any[])),i0.ɵmpd(512,i8.MatRippleModule,i8.MatRippleModule,
              ([] as any[])),i0.ɵmpd(512,i5.CommonModule,i5.CommonModule,([] as any[])),
          i0.ɵmpd(512,i8.MatPseudoCheckboxModule,i8.MatPseudoCheckboxModule,([] as any[])),
          i0.ɵmpd(512,i8.MatOptionModule,i8.MatOptionModule,([] as any[])),i0.ɵmpd(512,
              i9.PortalModule,i9.PortalModule,([] as any[])),i0.ɵmpd(512,i6.ScrollDispatchModule,
              i6.ScrollDispatchModule,([] as any[])),i0.ɵmpd(512,i7.OverlayModule,
              i7.OverlayModule,([] as any[])),i0.ɵmpd(512,i1.MatAutocompleteModule,
              i1.MatAutocompleteModule,([] as any[]))]);
    });
const styles_MatAutocomplete:any[] = ['.mat-autocomplete-panel{min-width:112px;max-width:280px;overflow:auto;-webkit-overflow-scrolling:touch;visibility:hidden;max-width:none;max-height:256px;position:relative}.mat-autocomplete-panel:not([class*=mat-elevation-z]){box-shadow:0 5px 5px -3px rgba(0,0,0,.2),0 8px 10px 1px rgba(0,0,0,.14),0 3px 14px 2px rgba(0,0,0,.12)}.mat-autocomplete-panel.mat-autocomplete-visible{visibility:visible}.mat-autocomplete-panel.mat-autocomplete-hidden{visibility:hidden}'];
export const RenderType_MatAutocomplete:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_MatAutocomplete,data:{}});
function View_MatAutocomplete_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,[[2,0],['panel',1]],(null as any),2,'div',
      [['class','mat-autocomplete-panel'],['role','listbox']],[[8,'id',0]],(null as any),
      (null as any),(null as any),(null as any))),i0.ɵdid(1,278528,(null as any),0,
      i5.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],
      {klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵncd((null as any),
      0)],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = 'mat-autocomplete-panel';
    const currVal_2:any = _co._classList;
    _ck(_v,1,0,currVal_1,currVal_2);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.id;
    _ck(_v,0,0,currVal_0);
  });
}
export function View_MatAutocomplete_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(402653184,1,{template:0}),i0.ɵqud(671088640,2,{panel:0}),
      (_l()(),i0.ɵand(0,[[1,2]],(null as any),0,(null as any),View_MatAutocomplete_1))],
      (null as any),(null as any));
}
export function View_MatAutocomplete_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,0,(null as any),(null as any),3,'mat-autocomplete',
      [['class','mat-autocomplete']],(null as any),(null as any),(null as any),View_MatAutocomplete_0,
      RenderType_MatAutocomplete)),i0.ɵdid(1,1097728,(null as any),2,i1.MatAutocomplete,
      [i0.ChangeDetectorRef,i0.ElementRef],(null as any),(null as any)),i0.ɵqud(603979776,
      1,{options:1}),i0.ɵqud(603979776,2,{optionGroups:1})],(null as any),(null as any));
}
export const MatAutocompleteNgFactory:i0.ComponentFactory<i1.MatAutocomplete> = i0.ɵccf('mat-autocomplete',
    i1.MatAutocomplete,View_MatAutocomplete_Host_0,{displayWith:'displayWith',classList:'class'},
    {optionSelected:'optionSelected'},['*']);
